src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java

```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
     public Book borrowBook(Long bookId, Long clientId) {
         Optional<Client> optionalClient = clientRepository.findById(clientId);
         Optional<Book> optionalBook = bookRepository.findById(bookId);
-        if (optionalClient.isPresent() && optionalBook.isPresent()) {
+        //added borrow validation
+        if (optionalClient.isPresent() && optionalBook.isPresent()&&optionalClient.get().getBooks().size()<2 ){
             Client client = optionalClient.get();
             Set<Book> bookSet = client.getBooks();
+            
             bookSet.add(optionalBook.get());
             client.setBooks(bookSet);
             clientRepository.save(client);
             return optionalBook.get();
         }
+        System.out.println("No books for u pilgrim");
         return null;
     }
 
     @Override
-    public List<Book> showBorrowedBooks() {
+    @Transactional
+    public Set<Book> showBorrowedBooks(Long clientId) {
+        Optional<Client> optionalClient = clientRepository.findById(clientId);
+        if(optionalClient.isPresent()){
+           
+            Client client=optionalClient.get();
+            return client.getBooks();
+        }
+        
+        else
+        System.out.println("No books");
+        
+        
+        
         return null;
```

Comment:
if client exists return empty collection, if doesn't throw exception and return appropriate http.status to client

Reply:
added